src/cljdoc/server/ingest.clj

```diff
@@ -37,13 +37,16 @@
 
 (defn ingest-git!
   "Analyze the git repository `repo` and store the result in `storage`"
-  [storage {:keys [project version scm-url local-scm pom-revision] :as _repo}]
+  [storage {:keys [project version scm-url pom-revision requested-revision] :as _repo}]
   {:pre [(string? scm-url)]}
-  (let [git-analysis (ana-git/analyze-git-repo project version (or local-scm scm-url) pom-revision)]
+  (let [scm-rev (or requested-revision pom-revision)
+        git-analysis (cond-> (ana-git/analyze-git-repo project version scm-url scm-rev)
+                       ;; The git tag expressing the project version is not relevant if git revision specifically requested
+                       requested-revision (update-in [:scm] dissoc :tag))]
```

Comment:
Could passing the revision twice here somehow be solved at the call site? It seems that the only reason we need `requested-revision` is to remove the `[:scm :tag]` field. Maybe we could use `(not= scm-rev (get ana-result [:scm :tag])` as a condition to remove it instead and this way avoid the extra arg?

Reply:
Probably.  I'll take a look.

Reply:
I can easily remove the extra arg but I'll keep the removal logic the same.

`:scm` -> `:tag` -> `:name` is preferred over `:scm` -> `:commit` [at url fixup time](https://github.com/cljdoc/cljdoc/blob/966b30ef8833143c97a7d700fd6e8ac9b265c5fc/src/cljdoc/util/fixref.clj#L77).
`:scm` -> `:tag` -> `:name` is the tag representing the project version, if it exists (ex "v1.2.3" or "1.2.3"), as far as I understand it.

I had a todo to check if this current preference logic is appropriate/consistent/documented but did not get to it.  My guess the intent was to provide a nicer fixed up URL (version looks nicer than git SHA), but I don't know.

Reply:
Oh wait... I see why I did what I did: `ingest-git!` analyzes AND stores, so I'm thinking, the way things are structured now... that extra arg is necessary to make the decision to turf `:scm` -> `:tag` after analysis and before storage.